### PR TITLE
Update python 2 to 3 (SOFTWARE-4786)

### PR DESCRIPTION
**Pull request by mtru32**

### CODE REVIEW

**Review by DrDaveD — CHANGES_REQUESTED** [review]
Thanks for doing this conversion!

The github workflow did not run.  I don't know the details of how to do that, but I assume something needs to be configured by a repo administrator to enable that.  I do not have administrator privileges.  @brianhlin can you enable it?

I noticed that the rpm spec file has not been updated to require python3.  Please test building rpms also in github actions.    [Here is an example](https://github.com/hpcng/singularity/blob/master/.github/workflows/ci.yml#L96) of how to do that.

**Comment by brianhlin** [comment]
Hi @DrDaveD, @mtru32's one of our students so I'd like to limit this PR to the Python-only changes and the GHA. We'll handle the RPM packaging in a separate PR.

WRT the GHA not running, all the repo settings look good. I think it's because the target branch has to have the workflow already merged in. @mtru32 would you be able to submit a PR for the Python2 GHA branch that you've got?

**Comment by brianhlin** [comment]
@DrDaveD ok, we had to have the GHA workflow in master before the updates to the workflow would run on this branch. Note that we don't have any tests for `osg-incommon-cert-request`.

**Review by DrDaveD — APPROVED** [review]

**Comment by DrDaveD** [comment]
Oh yeah there's an issue #53 for writing those unit tests, assigned to @jteheran but she never got around to it.

File: osgpkitools/cert_utils.py [hunk 1]
```diff
@@ -30,7 +30,7 @@ def get_ssl_context(usercert, userkey):
     pass_str = 'Please enter the pass phrase for'
     for _ in range(0, 2): # allow two password attempts
         def prompt_for_password(verify):
-            return getpass.getpass(pass_str+" '%s':" % userkey)
+            return getpass.getpass(pass_str+f" '{userkey}':")
```
Comment (brianhlin) [inline osgpkitools/cert_utils.py:33]:
```suggestion
            return getpass.getpass(pass_str + f" '{userkey}':")
```

Just a quick whitespace change for my own sanity

File: tests/OSGPKIUtilsTests.py [hunk 2]
```diff
@@ -18,26 +18,25 @@ class OSGPKIUtilsTests(unittest.TestCase):
...
-        self.assert_(cert.x509request, 'missing CSR contents')
+        self.assertTrue(cert.x509request, 'missing CSR contents')
 
     def test_alt_name_csr_generation(self):
         '''Generate a CSR with multiple SANs'''
         alias = 'test-san.' + DOMAIN
         cert = OSGPKIUtils.Cert(self.FQDN, altnames=[alias], email=EMAIL)
         csr_contents = cert.x509request.as_text()
-        self.assert_(re.search(r'X509v3 Subject Alternative Name: critical', csr_contents),
+        self.assertIsNone(re.search(r'X509v3 Subject Alternative Name: critical', csr_contents),
```
Comment (brianhlin) [inline tests/OSGPKIUtilsTests.py:28]: This seems to flip the test around. We want to make sure that we find this search string

File: tests/test_cert_request.py [hunk 3]
```diff
@@ -73,7 +73,7 @@ def test_required_opts(self):
                 with capture_sys_output() as (_, stderr):
                     self.assertRaises(SystemExit, parse_cli_flatten_args, args)
 
-                self.assert_(re.search(r'error.*is required.*', stderr.getvalue()))
+                self.assertIsNone(re.search(r'error.*is required.*', stderr.getvalue()))
```
Comment (brianhlin) [inline tests/test_cert_request.py:76]: Another case where I believe this flips the test

**Review by brianhlin — COMMENTED** [review]
Looks mostly good except for a few regex tests being swapped around.

The only other thing I see potentially being an issue is handling encoding where we [write files](https://github.com/opensciencegrid/osg-pki-tools/blob/ddd9706a2ddbb90a0832bd30a472e6e20dcc92ed/osgpkitools/utils.py#L18). One easy way to test this would be to check the contents of the generated CSR in the old version vs your Python 3 version. I'd be happy to merge this in for now and have you test that as part of https://opensciencegrid.atlassian.net/browse/SOFTWARE-4784

**Review by brianhlin — APPROVED** [review]
LGTM!

**Pull request merged**